File: Chapter_3/Fibonacci/C/main.c
#include <stdio.h>

// Nth fibonacci sequence. Using the simple recursive approach.
unsigned int get_nth_fibonacci(unsigned int n) {
    if (n == 0) {
        return 0;
    }

    if (n == 1) {
        return 1;
    }

    return get_nth_fibonacci(n - 1) + get_nth_fibonacci(n - 2);
}

int main(void) {
    for (unsigned int n = 0; n < 20; n++) {
        unsigned int fibonacci_value = get_nth_fibonacci(n);
        printf("%2d: %d\n", n, fibonacci_value);
    }
}

File: Chapter_3/Fibonacci/Rust/src/main.rs
// Return the nth value in the fibonacci sequence
fn get_nth_fibonacci_value(n: u32) -> u32 {
    if n == 0 {
        return 0;
    }

    if n == 1 {
        return 1;
    }

    return get_nth_fibonacci_value(n - 1) + get_nth_fibonacci_value(n - 2);
}

fn main() {
    for n in 0..20 {
        let nth_value = get_nth_fibonacci_value(n);
        println!("{}: {}", n, nth_value);
    }
}
